### ISSUE TITLE
Unable to Connect to F3800

### ISSUE THREAD

**Issue by BillyFKidney** [tiger2380/anker_f3800_web#2]
Synology Web Station 4.2.2-0498
PHP v8.0

The site loads but I am unable to connect to the F3800.
Pressing the CONNECT button does nothing, perhaps I missed a setup step?

<img width="1630" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/04f68b3c-20b9-433d-9232-74f5b3dd9fd4">
<img width="1115" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/33038805-081a-4a33-a3e5-956a2af0c158">

**Comment by tiger2380** [comment]
Hi. Which browser are you using? Also, when you click on "connect" does a selection box pop up and show a list of Bluetooth devices?

**Comment by BillyFKidney** [comment]
I was using Safari

**Comment by BillyFKidney** [comment]
> Hi. Which browser are you using? Also, when you click on "connect" does a selection box pop up and show a list of Bluetooth devices?

I've tested on MacOS 14.4.1 (23E224) and the behavior is the same, clicking connect does not pop up a list of devices:
Safari - Version 17.4.1 (19618.1.15.11.14)
Chrome - Version 124.0.6367.62 (Official Build) (arm64) 

I use Chrome often to program my ESP32 devices via USB. I've never tried to use a browser to connect to a Bluetooth device.
<img width="646" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/96e4c932-da88-485e-9363-41ad2566be70">

**Comment by BillyFKidney** [comment]
I found the issues:
1) Safari does not appear to allow you to connect to Bluetooth devices.
2) Synology wasn't using HTTPS to host the site, which is why I wasn't able to connect in Chrome.

I have connected to the F3800 in Chrome successfully
<img width="445" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/c0887603-655b-42e4-ba84-24b1e8a7148a">

However all of the graphs still appear blank:
<img width="1057" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/d54e5dc0-bbed-4ce5-b851-f33909e127b4">

Which should not be the case:
![image](https://github.com/tiger2380/anker_f3800_web/assets/65253277/f5bd3199-1e9e-48b3-a83c-c45a8c664d28)


![image](https://github.com/tiger2380/anker_f3800_web/assets/65253277/4e0571f4-4d01-4ab4-8cde-9e7b1a1dbaac)

**Comment by tiger2380** [comment]
Interesting. Are there any errors in the browser's console?

**Comment by BillyFKidney** [comment]
I'm not seeing any errors...after pressing CONNECT in the browser do I need to do anything on the F3800?
<img width="552" alt="image" src="https://github.com/tiger2380/anker_f3800_web/assets/65253277/baf487d7-24f3-4ec3-bc9a-24cda58d45e9">

**Comment by BillyFKidney** [comment]
Any ideas?

Did I miss a step on the device (F3800)?

**Comment by tiger2380** [comment]
Sorry for the late response.  Strange. My arraybuffer is only 240ish bytes. I see that yours is more than 400 bytes. So I filtered to only look for 240 bytes to keep out garbage.  I need to improve in my filtering. Do you have an expansion battery connected?

**Comment by BillyFKidney** [comment]
Yes, I have one expansion battery.On May 14, 2024, at 6:14 PM, Thaddeus Bibbs ***@***.***> wrote:﻿
Sorry for the late response.  Strange. My arraybuffer is only 240ish bytes. I see that yours is more than 400 bytes. So I filtered to only look for 240 bytes to keep out garbage.  I need to improve in my filtering. Do you have an expansion battery connected?

**Comment by tiger2380** [comment]
Try disconnecting the battery and try again. Sorry, I don't have an expansion battery and not able to test it.

**Comment by BillyFKidney** [comment]
> Try disconnecting the battery and try again. Sorry, I don't have an expansion battery and not able to test it.

Sorry for the lag, I disconnected the battery and it connected successfully.
![image](https://github.com/tiger2380/anker_f3800_web/assets/65253277/00859bea-592a-4559-98c4-6f7591915914)

However, reconnecting the external battery caused the data to disappear. Is there anything I can do to assist supporting the external battery?
